configure.ac

```diff
@@ -1351,8 +1351,6 @@ HWLOC_DO_AM_CONDITIONALS
 AM_CONDITIONAL([have_hwloc], [test "${have_hwloc}" = "yes"])
 AM_CONDITIONAL([use_embedded_hwloc], [test "${use_embedded_hwloc}" = "yes"])
 
-AM_CONDITIONAL([HAVE_HWLOC], [test x$have_hwloc = xyes])
```

Comment:
This should not be removed as `src/mpi` also needs it

Reply:
`HAVE_HWLOC` is still defined. See https://github.com/pmodels/mpich/blob/21f9c56aae5631bd2a6d096f788846b102872e1d/configure.ac#L1347
Removing this line means we cannot use `HAVE_HWLOC` macro in any `Makefile`, which is fine because only topology aware trees was doing it anyway.

Reply:
I see, then I think the PR is fine.